tests/optim/test_oss.py

```diff
@@ -234,6 +235,7 @@ def closure():
     optimizer.load_state_dict(optimizer_state_dict)
 
 
+@skip_if_not_enough_gpu
```

Comment:
is that strictly needed ? we can have several processes storing data to the same gpu, no ? I'm a bit surprised that this is needed actually

Reply:
It appears to be needed. It fails on my system with only 2 GPUs. But I didn't debug deeply into it. Here is more info:

```
>       raise Exception(msg)
E       Exception:
E
E       -- Process 2 terminated with the following error:
E       Traceback (most recent call last):
E         File "/scratch/m1n/env1/lib/python3.6/site-packages/torch/multiprocessing/spawn.py", line 20, in _wrap
E           fn(i, *args)
E         File "/private/home/m1n/git/fairscale/tests/optim/test_oss.py", line 197, in run_test_collect_shards
E           target = torch.rand((batch, target_width), device=device)
E       RuntimeError: CUDA error: invalid device ordinal

/scratch/m1n/env1/lib/python3.6/site-packages/torch/multiprocessing/spawn.py:119: Exception
```

When I print the `device` param above, here is what I see:

```
tests/optim/test_oss.py::test_collect_shards cuda:1
cuda:2
cuda:0
FAILED
```

It seems to hardcode to 3 GPUs?